packages/runtime-core/src/vnode.ts

```diff
@@ -443,10 +443,8 @@ const normalizeRef = ({
     ref = '' + ref
   }
   return (
-    ref != null
-      ? isString(ref) || isRef(ref) || isFunction(ref)
-        ? { i: currentRenderingInstance, r: ref, k: ref_key, f: !!ref_for }
-        : ref
+    ref != null && (isString(ref) || isRef(ref) || isFunction(ref))
```

Comment:
```suggestion
    (isString(ref) || isRef(ref) || isFunction(ref))
```
`ref != null` seems no longer needed.

Reply:
True!